libgstd/gstd_http.c

```diff
@@ -385,14 +387,15 @@ server_callback (SoupServer * server, SoupMessage * msg,
   }
   data_request->mutex = &self->mutex;
 
-  soup_message_headers_append (msg->response_headers,
+  response_headers = soup_server_message_get_request_headers (msg);
+  soup_message_headers_append (response_headers,
       "Access-Control-Allow-Origin", "*");
-  soup_message_headers_append (msg->response_headers,
+  soup_message_headers_append (response_headers,
       "Access-Control-Allow-Headers", "origin,range,content-type");
-  soup_message_headers_append (msg->response_headers,
+  soup_message_headers_append (response_headers,
       "Access-Control-Allow-Methods", "PUT, GET, POST, DELETE");
   g_mutex_lock (&self->mutex);
-  soup_server_pause_message (server, msg);
+  soup_server_message_pause (msg);
```

Comment:
This change was added since libsoup 3.2, for versions minor to 3.2 it will fail as it expect to use soup_server_pause_message(). There are some distributions that manage libsoup 3.0.7 version like Ubuntu 22.04. For this case, we will need to add conditional logic to support versions < 3.2 and >= 3.2.